File: ProcessingService/app/models/order.py
from pydantic import BaseModel
from typing import Optional
from enum import Enum
from datetime import datetime

class OrderStatus(str, Enum):
    RECEIVED = "received"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"

class Order(BaseModel):
    order_id: str
    status: OrderStatus
    created_at: datetime
    processed_at: Optional[datetime] = None
    error_message: Optional[str] = None

File: OrderService/Models/Order.cs
using System.Text.Json.Serialization;

namespace OrderService.Models;

public class Order
{
    [JsonPropertyName("order_id")]

    public string OrderId { get; set; } = default!;
    public OrderStatus Status { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? ProcessedAt { get; set; }
    public string? ErrorMessage { get; set; }
}